11-bug_tracking.Rmd

```diff
@@ -1,23 +1,125 @@
+
+## What may not be a bug?
+
+1. In case the code is doing something unexpected, it may not necessarily be a bug. Carefully review the documentation of the function being called, and check whether the behaviour being exhibited on calling this function is the same as it was designed to do.
+
+2. Issues with $\textit{seemingly}$ identical numbers not being equal (especially floating point numbers) are usually not bugs. 
+
+3. If $\textsf{R}$ is running slower than expected, then also it may not be a bug. Ask someone else to review your code in such a case.
+
+4. If some function is working, but it is not defined in the best generalised way, then consult some to look over your code. This may perhaps not be a bug, instead it might be an alternative way of writing the function.
+
+## Checking if a bug is already reported
+
+The first step before filing a bug report is to see whether the problem has already been reported. Checking if the bug is reported will:
+
+1. Save time for you and the developers
+
+2. Help you see if the bug is already fixed for the next release
+
+3. Lead you to learn what needs to be done to fix it
+
+4. Determine if any additional information is needed
+
+The sections that follow discuss where to check whether a bug is already reported.
+
+## Levels of contributing to bug / What do you do when you find a bug?
+
+1. [Report](#ReportBug) the bug (if it is not alreasy reported).
+
+2. [Test](provide hyperlink to the section/chapter which discusses this) the bug.
+
+3. [Fix](hyperlink to the chapter on the lifecycle of a patch) the bug: Fixing a bug might require relatively more time. You may start a conversation about it either on BugZilla (link here) or on the GitHub mirror (link here). This would require engagement with $\textsf{R}$ Core team.
+
+## What are some places where you may find a bug?
+
+You may find a bug in:
+
+1. The [documentation of this guide](#GuideDocBug).
+
+2. In the [$\textsf{R}$ Core-supported packages and/or their documentations](#RCorePkgBug).
+
+3. In the [$\textsf{R}$ language](#RLangBug).
+
+4. In [packages and/or their documentations which are not supported by the $\textsf{R}$ Core](#nonRCorePkgBug).
+
+## How to report a bug? {#ReportBug}
+
+Once you are confirmed that there is a bug, you need to submit a report so that it gets fixed.
+
+### Bug in the documentation of this guide {#GuideDocBug}
+
+1. If there is a bug in the documentation of this guide, then please report it as an issue at https://github.com/forwards/rdevguide/issues. 
+
+2. Please go through the $\textit{closed}$ issues to check if the bug is already reported and fixed. Otherwise, please check in the $\textit{open}$ issues to see if it is reported. If it is not recorded at either of these places, then please submit a new issue.

```

Comment:
In the introduction or as the last of the four sub-sub-sections of "Bug in ..."? What would be less confusing to readers? What do you suggest?

Reply:
I think that improvements on the book itself should be addressed on the introduction of the book (so on index.Rmd itself). It will be a great way to encourage more participation of people on the book.

Reply:
Ok. My plan is to write the introduction to the book chapter after some chapters are completed. Yes, it sounds good to include the improvements to the book and reporting bugs in this documentation, in that chapter.